### PR TITLE
Make babel-runtime use regenerator-runtime from npm

### PR DESCRIPTION
**Pull request by benjamn**
The source repository for `regenerator-runtime` can now be found at https://github.com/facebook/regenerator/tree/master/packages/regenerator-runtime. Note that `regenerator-runtime` is now published as an independent package, so `babel-runtime` no longer needs to pull in all of `regenerator` just to depend on the runtime.

This also removes the `babel-regenerator-runtime` package, now that the only difference between it and `regenerator-runtime` has been resolved: https://github.com/facebook/regenerator/commit/bc8d2d672ce768726e593a3bfad2cb18b85f9547

It's also my plan to move https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-regenerator back into the https://github.com/facebook/regenerator repository as an independent package, but I'll do that in a separate pull request.

I'm happy to work with the Babel team to keep the workflow as smooth as possible, but if I'm going to spend time addressing Regenerator bug reports, then https://github.com/facebook/regenerator really needs to be the source of truth, since that's where all the tests are, and that's where people (especially those not using Babel!) go to report bugs.

### CODE REVIEW

File: packages/babel-runtime/package.json [hunk 1]
```diff
@@ -6,12 +6,12 @@
   "repository": "https://github.com/babel/babel/tree/master/packages/babel-runtime",
   "author": "Sebastian McKenzie <sebmck@gmail.com>",
   "dependencies": {
-    "core-js": "^2.1.0"
+    "core-js": "^2.1.0",
+    "regenerator-runtime": "^0.9.5"
```
Comment (benjamn) [inline packages/babel-runtime/package.json]: Should this be an exact version constraint? I don't want minor `regenerator-runtime` changes to risk breaking `babel-runtime`.

**Comment by loganfsmyth** [comment]
All for using the main regenerator repo if it's easy, I'm not familiar with what drove the current duplication.

I don't quite see why we'd want to move the regenerator transform our of the Babel repo though.

**Comment by benjamn** [comment]
> I don't quite see why we'd want to move the regenerator transform our of the Babel repo though.

Some reasons:

1. The current duplication was mostly motivated by Babel's desire to port the Regenerator transform to a Babel transformer style, whereas the original was based on [`ast-types`](https://github.com/benjamn/ast-types/#ast-traversal). I'm offering to use the Babel style instead of the `ast-types` style, going forward, which should make the duplication much less compelling.
2. Having `regenerator-runtime` and `regenerator-transform` in the same repository will eliminate any confusion about which versions of the transformer are compatible which versions of the runtime. Keeping them separate makes it all too easy for me to release a new version of the runtime that subtly breaks packages that depend on both. This has happened several times before, in case you don't remember.
3. I'm the original author and by far the most prolific maintainer of Regenerator, and I'm offering to keep working on it, on the condition that we use a single codebase. I would love to merge pull requests, fix issues, and add new features, but getting those changes ported over to Babel after the fact is a lot more work than fixing them once. I hope you can hear the realism in this plea, rather than taking it as an ultimatum. Help me help you!
4. You can still fork the https://github.com/facebook/regenerator codebase if you need to, but the hope is that any incremental changes Babel makes in its fork will be more easily upstreamable (and rebasing against upstream will be much more straightforward).

**Comment by benjamn** [comment]
Another big reason: Babel's fork of Regenerator does not run Regenerator's [extensive](https://github.com/facebook/regenerator/blob/master/test/tests.es6.js) [test](https://github.com/facebook/regenerator/blob/master/test/async.es6.js) [suite](https://github.com/facebook/regenerator/blob/master/test/tests.transform.js), which makes me uneasy, given the complexity of the transform.

Just this morning, I tried running Regenerator using Node 6, only to discover that Node's implementation of `Generator.prototype.return` is [broken](https://github.com/nodejs/node/issues/6525) (and Regenerator gets it right). Would the Babel team have noticed that discrepancy? What if the error had been on Regenerator's side, rather than Node's? Moving the code back to the main Regenerator repository is the easiest way to restore all those tests.

Fun observation: this pull request (#3494) actually increased Babel's test coverage by a tiny amount, even though I didn't add any additional tests… because I removed untested code. And that's a good thing! I would be happy to take responsibility for testing the transform and the runtime, if you'll let me.

**Comment by loganfsmyth** [comment]
Sorry, this was a misunderstanding on my part. I was thinking that our fork of `regenerator` and `transform-regenerator` were two separate modules already, I wasn't trying to imply we shouldn't move it, I was trying to understand the whether moving the actual Babel plugin made the most sense, or if we should have basically a stub plugin that called out to regenerator in the main babel repo still.

My main concern is keeping the versioning consistent and I also feeling slightly weird about having our official `es2015` preset depend directly on a transform that didn't live in our repo, even if `regenerator` itself is separate, but that may just be something I need to not worry about :P

**Comment by benjamn** [comment]
Thanks for the clarification. I'd be totally happy for Babel to keep `babel/packages/babel-plugin-transform-regenerator` as a shell that depends on an exact version of `regenerator-transform` (as published from the https://github.com/facebook/regenerator repo). Then I can submit a Babel pull request to bump that version whenever I ship a new version of `regenerator-transform`, and you can decide whether/when to merge it. A similar protocol (exact version + PR to bump it) worked well for stopping transform/runtime mismatches in the past.

**Comment by hzoo** [comment]
I'm cool with this. The more help we can get the better, esp. if it's @benjamn for regenerator?

File: packages/babel-runtime/package.json (hunk 1 above)
Comment (hzoo) [inline packages/babel-runtime/package.json]: Yeah I guess so, and we would manually update babel-runtime when regenerator changes?

Comment (benjamn) [inline packages/babel-runtime/package.json]: Yeah, I'm happy to submit PRs to bump both versions.

**Comment by hzoo** [comment]
@kittens is cool with just using facebook/regenerator too

**Comment by benjamn** [comment]
Rebased and resolved conflicts.

**Comment by hzoo** [comment]
I think the last thing is just updating the docs/website again.

**Pull request merged**